src/api.ts

```diff
@@ -99,34 +101,75 @@ export async function deleteImage(hash: CommitHash) {
 		l.error({ err, commitHash: hash }, 'failed to remove image');
 	}
 }
-
 export async function startContainer(commitHash: CommitHash) {
 	l.log({ commitHash }, `Starting up container for ${commitHash}`);
 	const image = getImageName(commitHash);
+
+	// do we have an existing container?
+	const existingContainer = getRunningContainerForHash( commitHash );
+	if ( existingContainer ) {
+		l.log( { commitHash, containerId: existingContainer.Id }, `Found a running container for ${commitHash}`);
+		return Promise.resolve( existingContainer );
+	}
+
+	// are we starting one already?
+	if( state.startingContainers.has( commitHash ) ) {
+		l.log( { commitHash, containerId: existingContainer.Id }, `Already starting a container for ${commitHash}`)
```

Comment:
should we have `existingContainer.Id` here if we didn't execute in the branch above testing `if ( existingContainer )`? seems like it would be `running` and already return or `null`

Reply:
ah good catch

Reply:
fixed